src/terms/hubbard.jl

```diff
@@ -8,41 +8,58 @@ Structure for Hubbard manifold choice and projectors extraction.
 to the set of atomic orbitals used to compute the Hubbard correction.
 It is to be noted that this is not meant in the mathematical sense of "manifold".
 
-Overview of fields:
-- `psp`: Pseudopotential containing the atomic orbital projectors
-- `iatoms`: Atom indices that are part of the manifold.
-- `l`: Angular momentum quantum number of the manifold.
-- `i`: Index of the atomic orbital within the given l.
-
-See also the convenience constructors, to construct a manifold more easily.
+The manifold contains two pieces of information:
+- The atomic sites at which the correction is applied.
+  These can either be specified as a chemical species for convenience,
+  or as a list of atom indices for maximum flexibility.
+- The pseudoatomic orbitals to use for the projectors.
+  These can either be specified by their label (e.g. `"3D"` for PseudoDojo),
+  or by their angular momentum number `l` + the corresponding index `i`.
 """
 struct OrbitalManifold
+    atoms::Union{ChemicalSpecies, Vector{Int64}}
+    projectors::Union{AbstractString, @NamedTuple{l::Int64, i::Int64}}
+end
+function OrbitalManifold(atom::ElementPsp, projectors)
+    OrbitalManifold(atom.species, projectors)
+end
+function OrbitalManifold(atom::Symbol, projectors)
+    OrbitalManifold(ChemicalSpecies(atom), projectors)
+end
+
+"""
+DFTK-internal version of [`OrbitalManifold`](@ref)
+with resolved atom indices and projectors.
+
+This separation allows the manifold to be defined
+without requiring access to the pseudopotential.
+"""
+struct ResolvedOrbitalManifold
     psp::NormConservingPsp
     iatoms::Vector{Int64}
     l::Int64
     i::Int64
 end
-function OrbitalManifold(atoms::Vector{<:Element}, atom::ElementPsp, label::AbstractString)
-    OrbitalManifold(atom, findall(at -> at === atom, atoms), label)
-end
-function OrbitalManifold(atom::ElementPsp, iatoms::Vector{Int64}, label::AbstractString)
-    OrbitalManifold(atom.psp, iatoms, label)
-end
-function OrbitalManifold(psp::NormConservingPsp, iatoms::Vector{Int64}, label::AbstractString)
-    (; l, i) = find_pswfc(psp, label)
-    OrbitalManifold(psp, iatoms, l, i)
-end
 
-function check_hubbard_manifold(manifold::OrbitalManifold, model::Model)
-    for atom in model.atoms[manifold.iatoms]
+function resolve_hubbard_manifold(manifold::OrbitalManifold, model::Model)
+    if manifold.atoms isa ChemicalSpecies
```

Comment:
Do the reverse and check for abstractvector of integers (and be general in the checked type)